frontend/src/devicelist/__tests__/ChannelDetailsTableRow.test.jsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import Enzyme from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+
+import { afterEach, describe, expect, jest } from "@jest/globals";
+import { TableCell, TableRow } from "@material-ui/core";
+import InputChannelInfo from "../../model/InputChannelInfo";
+import OutputChannelInfo from "../../model/OutputChannelInfo";
+import ChannelDetailsTableRow from "../ChannelDetailsTableRow";
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe("ChannelDetailsTableRow", () => {
+  let wrapper;
+  const dummyId = 69;
+  const dummyName = "TEST_NAME";
+  const dummyPort = 1231;
+  const inChannel = new InputChannelInfo(dummyId, dummyName, dummyPort, null);
+  const outChannel = new OutputChannelInfo(dummyId, dummyName, dummyPort, null);
```

Comment:
since in & out channels have the same details & the tests don't care about whether the channel is for input or output then we can probably just remove this to lose the smell

Reply:
I'm adding test to check for invalid inputs